### PR TITLE
1367 validate file size

### LINKED ISSUE #1367: Add validation state for supporting file that is too large
**Describe the bug**
When attempting to upload an invalid file, e.g. one more than 100MB, there is no indication why it cannot be uploaded. The file manager closes with the UI unchanged.

**Steps To Reproduce**
1. Navigate to the file page.
2. Upload a manuscript successfully.
3. Click the link to upload supporting file(s).
4. Observe that file manager closes with no file added.

**Expected behavior**
We should feed back to the user that the file they have attempted to upload is invalid. They should then be able to clear this validation message by deleting the item from the list.

Design:

screen shot 2019-01-22 at 12 46 23

We should cap the number of items displayed to the user at 10, regardless of whether the items are valid or invalid.

The user should be able to continue through the wizard while invalid files are listed.

The user can also re-upload a file with the same name to override the error. (<- How difficult is this techinically? We could potentially drop for launch)

### PR DESCRIPTION
**Pull request by SemraSh**
#### Background
Adds validation for supporting files larger than 100mb

#### Any relevant tickets
Closes #1367

### CODE REVIEW

**Comment by diversemix** [comment]
I tested this in the review environment - looks good.

Minor point but the text starts with a capital 'M' and not a dash and lower case as per the spec... but I'm sure you know that anyway :wink:

**Comment by SemraSh** [comment]
Confirmed with Chris that the design is as implemented in this pr.

**Review by jyenes — APPROVED** [review]
we have created a test browser for this but GitHub doesn't allow us to upload files larger than 100 MB

File: app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js [hunk 1]
```diff
@@ -185,40 +239,32 @@ class SupportingUpload extends React.Component {
...
+                {file.success && <UploadSuccessIcon />}
+                {(file.error || file.rejected) && <UploadFailureIcon />}
+              </IconHolder>
+              <FileHolder>
+                <FileName data-test-id="file-block-name">
+                  {file.file.name ? file.file.name : file.file.filename}
+                </FileName>
+                {file.rejected && (
+                  <ErrorMessage data-test-id="file-block-error">
+                    Must be less than 100mb
```
Comment (giorgiosironi) [inline app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js]: should read `config.fileUpload.maxSizeMB`?

**Review by diversemix — APPROVED** [review]

**Comment by hdrury1** [comment]
Meets Product reqs. At the moment refreshing the page removes the error but I presume this will be addressed when tackling #1362

**Review by hem-brahmbhatt — APPROVED** [review]
Code looks good

**Comment by jyenes** [comment]
> Meets Product reqs. At the moment refreshing the page removes the error but I presume this will be addressed when tackling #1362

@hdrury1 why do have to persist an invalid submission?
I mean, the file wasn't uploaded to S3, and also wasn't persisted in the database, is just a warning to show to the user, from my point of view, we shouldn't persist this status in the database. am I wrong?

**Comment by hdrury1** [comment]
Hi @jyenes yes I think this is the case (at least right now, see https://elifesciences.slack.com/archives/C6LKMFK5Z/p1548337091016000), and design is happy with it. Good to merge!

**Pull request merged**